source/module_hsolver/hsolver_pw.cpp

```diff
@@ -208,12 +209,45 @@ void HSolverPW<T, Device>::paw_func_after_kloop(psi::Psi<T, Device>& psi, elecst
 
 #endif
 
+template <typename T, typename Device>
+void HSolverPW<T, Device>::cal_ethr_band(const double& wk, const double* wg, const double& ethr, std::vector<double>& ethrs)
+{
+    if(wk > 0.0)
+    {
+        // Note: the idea of threshold for unoccupied bands (1e-5) comes from QE
+        // In ABACUS, We applied a smoothing process to this truncation to avoid abrupt changes in energy errors between different bands.
+        const double ethr_unocc = std::max(1e-5, ethr);
```

Comment:
there are three magic numbers, 1e-5, 1e-2, 1e-5 in these lines, could you use three variables to store these values in front of the code if(wk>0.0), and use these variables instead?